### ISSUE TITLE
[no-output-native] incorrectly throws error if typing-names matches forbidden event names

### ISSUE THREAD

**Issue by stianmorsund** [angular-eslint/angular-eslint#543]
**Description and reproduction of the issue**

Given e.g the following output, where the actual typings have the words success/error (which are part of the native forbidden event names) will give a linting error.

```
@Output() save: EventEmitter<(success?: boolean, error?: boolean) => void> = new EventEmitter();
```

Doing e.g the following will pass, but seems kind of verbose:

```
type CallbackFunction = (success?: boolean, error?: boolean) => void;
@Output() save: EventEmitter<CallbackFunction> = new EventEmitter();
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/eslint-plugin`    | `12.1.0` |
| `@typescript-eslint/parser`        | `4.19.0` |
| `ESLint`                           | `7.28.0` |
| `node`                             | `14.16.1` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
Angular CLI: 12.0.4
Node: 14.16.1
Package Manager: yarn 1.22.10
OS: darwin x64

Angular: 12.0.4
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, language-service, localize, material
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1200.4
@angular-devkit/build-angular   12.0.4
@angular-devkit/core            8.3.23
@angular-devkit/schematics      8.3.23
@angular/flex-layout            12.0.0-beta.34
@schematics/angular             12.0.4
rxjs                            6.6.7
typescript                      4.2.4

```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by stianmorsund** [comment]
Seems like this should be fixed by https://github.com/angular-eslint/angular-eslint/pull/524

Just needs a new release :)

**Comment by rafaelss95** [comment]
Yes, as you mentioned it's already fixed on master, so it'll be available in the next version.
